test/bench/grpc_bench_test.go

```diff
@@ -0,0 +1,291 @@
+		)
+		assert.NoError(b, err)
+
+		err = c.Activate(context.Background())
+		assert.NoError(b, err)
+
+		clients = append(clients, c)
+	}
+	return
+}
+
+// cleanupClients is a helper function to clean up clients.
+func cleanupClients(b *testing.B, clients []*client.Client) {
+	for _, c := range clients {
+		assert.NoError(b, c.Deactivate(context.Background()))
+		assert.NoError(b, c.Close())
+	}
+}
+
+func benchmarkUpdateAndSync(cnt int, ctx context.Context, b *testing.B, cli *client.Client, d *document.Document, key string) error {
+	for i := 0; i < cnt; i++ {
+		err := d.Update(func(root *proxy.ObjectProxy) error {
+			text := root.GetText(key)
+			text.Edit(0, 0, "c")
+			return nil
+		})
+		assert.NoError(b, err)
+		err = cli.Sync(ctx)
+		assert.NoError(b, err)
+	}
+	return nil
+}
+
+func benchmarkUpdateProject(cnt int, adminCli *admin.Client, b *testing.B) error {
+	ctx := context.Background()
+	for i := 0; i < cnt; i++ {
+		name := fmt.Sprintf("name%d", i)
+		authWebhookURL := fmt.Sprintf("authWebhookURL%d", i)
+		authWebhookMethods := []string{}
+		for _, m := range types.AuthMethods() {
+			authWebhookMethods = append(authWebhookMethods, string(m))
+		}
+
+		_, err := adminCli.UpdateProject(
+			ctx,
+			"000000000000000000000000",
+			&types.UpdatableProjectFields{
+				Name:               &name,
+				AuthWebhookURL:     &authWebhookURL,
+				AuthWebhookMethods: &authWebhookMethods,
+			},
+		)
+		assert.NoError(b, err)
+	}
+	return nil
+}
+
+func watchDoc(cli *client.Client, rch <-chan client.WatchResponse, done <-chan bool, ctx context.Context, b *testing.B, wg *sync.WaitGroup) {
+	defer wg.Done()
+	for {
+		select {
+		case resp := <-rch:
+			if resp.Err == io.EOF {
+				assert.Fail(b, resp.Err.Error())
+			}
+			assert.NoError(b, resp.Err)
+
+			if resp.Type == client.DocumentsChanged {
+				err := cli.Sync(ctx, resp.Keys...)
+				assert.NoError(b, err)
+			}
+		case <-done:
+			return
+		}
+	}
+}
+
+func BenchmarkRPC(b *testing.B) {
+	b.StopTimer()
+	logging.SetLogLevel("error")
+	startDefaultServer()
+	defer func() {
+		if defaultServer != nil {
+			if err := defaultServer.Shutdown(true); err != nil {
+				logging.DefaultLogger().Error(err)
+			}
+		}
+	}()
+	b.StartTimer()
+	b.Run("client to server", func(b *testing.B) {
+		b.StopTimer()
+		for i := 0; i < b.N; i++ {
+			cli, err := client.Dial(
+				defaultServer.RPCAddr(),
+				client.WithPresence(types.Presence{"name": fmt.Sprintf("name-%s", b.Name())}),
+			)
+			assert.NoError(b, err)
+			defer func() {
+				err := cli.Close()
+				assert.NoError(b, err)
+			}()
+
+			ctx := context.Background()
+			err = cli.Activate(ctx)
+			assert.NoError(b, err)
+			assert.True(b, cli.IsActive())
+
+			d1 := document.New("doc1")
+			err = cli.Attach(ctx, d1)
+			assert.NoError(b, err)
+
+			testKey := "testKey"
+			err = d1.Update(func(root *proxy.ObjectProxy) error {
+				root.SetNewText(testKey)
+				return nil
+			})
+			assert.NoError(b, err)
+
+			b.StartTimer()
```

Comment:
This code doesn't seem to work as intended. Let's remove b.StopTimer and b.StartTimer for now.

```suggestion
			b.StartTimer()
```